example/number.eag

```diff
@@ -0,0 +1,25 @@
+// A simple example referenced in the tutorial for showing how the optional EBNF operator works
```

Comment:
Yes, maybe a better example.
I have to think about and check against the tutorial text.

Reply:
Not worked on yet, until commit https://github.com/linkrope/gamma/pull/10/commits/c71be10dd828adcd430a3eb710845b86b4fe0d33

Reply:
@linkrope 

What do you think about the following two example specifications?
For option operator:
```
Expr<+ Sign>: <Sign>[ <+'Plus':Sign> '+' | <+'Minus':Sign> '-' ]<+'Plus':Sign> Term.
Sign = 'Plus' | 'Minus'.
Term: 'x'.
```

For grouping operator:
```
ExprG<+ Op>: Term <Op>( <+'Addition':Op> '+' | <+'Subtraction':Op> '-' ) Term.
Op = 'Addition' | 'Subtraction'.
Term: 'x'.
```

I would name the examples, _ebnf-option.eag_ and _ebnf-grouping.eag_ as this are really minimal examples an quite meaningless otherwise.

Reply:
Even if it's "+" or "-" in both cases, it's better to distinguish between "sign" and "op":
one next addition for "op" could be "OR", which does not make sense for "sign";
and one possbility for "sign" could be <empty>, which does not make sense for "op".

"op" and "sign" will go different ways in more realisitic examples, so we should keep them separate from the beginning.

BTW: replacing the missing sign with "+" only works with numbers, but not with booleans or strings. The best option for the sign would be to put the term into `[...]`, and to get it out unchanged, if there is no sign: `<- Term :Term, + Term: Expr>`. However, this would not work wit left-to-right evaluation. The second best option for the sign is the additional empty alternative. A "sign" is "+" or "-" or not there at all.

Reply:
> Even if it's "+" or "-" in both cases, it's better to distinguish between "sign" and "op":
one next addition for "op" could be "OR", which does not make sense for "sign";
and one possbility for "sign" could be , which does not make sense for "op".
>"op" and "sign" will go different ways in more realisitic examples, so we should keep them separate from the beginning.

Added the 2 simple, meaningless examples, one with "sign", one with "op", both lower case.